Workbooks/Azure Web Application Firewall/WAF monitoring/WAF Monitoring.workbook

```diff
@@ -0,0 +1,831 @@
+            "crossComponentResources": [
+              "value::selected"
+            ],
+            "typeSettings": {
+              "additionalResourceOptions": [
+                "value::all"
+              ],
+              "showDefault": false
+            },
+            "timeContextFromParameter": "TimeRange",
+            "queryType": 1,
+            "resourceType": "microsoft.resourcegraph/resources",
+            "value": [
+              ""
+            ]
+          },
+          {
+            "id": "49e2f511-592f-4d7f-8fda-d686803f3dbf",
+            "version": "KqlParameterItem/1.0",
+            "name": "TimeRange",
+            "label": "Time Range",
+            "type": 4,
+            "isRequired": true,
+            "value": {
+              "durationMs": 5707080000,
+              "endTime": "2023-01-06T18:50:00.000Z"
+            },
+            "typeSettings": {
+              "selectableValues": [
+                {
+                  "durationMs": 300000
+                },
+                {
+                  "durationMs": 900000
+                },
+                {
+                  "durationMs": 1800000
+                },
+                {
+                  "durationMs": 3600000
+                },
+                {
+                  "durationMs": 14400000
+                },
+                {
+                  "durationMs": 43200000
+                },
+                {
+                  "durationMs": 86400000
+                }
+              ],
+              "allowCustom": true
+            }
+          },
+          {
+            "id": "604a42a0-deca-4a95-a15f-8977646a7fac",
+            "version": "KqlParameterItem/1.0",
+            "name": "WAFType",
+            "type": 5,
+            "isRequired": true,
+            "multiSelect": true,
+            "quote": "'",
+            "delimiter": ",",
+            "query": "let FakeData = (datatable (Message:string, ruleName_s:string, clientIp_s:string, clientIP_s:string, action_s:string, transactionId_s:string,trackingReference_s:string) [ \"\", \"\", \"\", \"\", \"\", \"\", \"\" ]);\r\nFakeData | union AzureDiagnostics\r\n| where ResourceType == \"APPLICATIONGATEWAYS\" or ResourceType == \"FRONTDOORS\" or ResourceType == \"PROFILES\" or ResourceType == \"CDNWEBAPPLICATIONFIREWALLPOLICIES\"\r\n| summarize Count=count() by ResourceType\r\n| extend ResourceTypeImproved = iif(ResourceType == \"APPLICATIONGATEWAYS\", \"Application Gateway\", ResourceType)\r\n| extend ResourceTypeImproved = iif(ResourceTypeImproved == \"FRONTDOORS\", \"Azure Front Door\", ResourceTypeImproved)\r\n| extend ResourceTypeImproved = iif(ResourceTypeImproved == \"PROFILES\", \"Azure Front Door Premium\", ResourceTypeImproved)\r\n| extend ResourceTypeImproved = iif(ResourceTypeImproved == \"CDNWEBAPPLICATIONFIREWALLPOLICIES\", \"Azure CDN\", ResourceTypeImproved)\r\n| order by Count desc, ResourceTypeImproved asc\r\n| project ResourceTypeImproved",
```

Comment:
if i'm interpreting this correctly, i think you could dramatically simplify this and every query below.

dropdowns support 2 columns of data:
1. a "value" (like "APPLICATIONGATEWAYS") generally used *in queries*
2. a "label" (like "Application Gateway") generally displayed *to users*

you could have this dropdown return 2 columns like that, and then when a user selects the "Application Gateway" item, the parameter value would be "APPLICATIONGATEWAYS"

(if there are fixed set of them that you know up front, you can list them a different way statically here and you don't even have to run a query at all (unless you want them to be filtered to only the existing values?)

if you do that, ALL of the queries below that have this big block:
```
| where (ResourceType == "APPLICATIONGATEWAYS" or ResourceType == "FRONTDOORS" or ResourceType == "PROFILES" or ResourceType == "CDNWEBAPPLICATIONFIREWALLPOLICIES") and ("{WAFType:label}" == "All" or (ResourceType == "APPLICATIONGATEWAYS" and "{WAFType:label}" contains "application gateway") or (ResourceType == "FRONTDOORS" and  "{WAFType:label}" contains "azure front door") or (ResourceType == "PROFILES" and "{WAFType:label}" contains "azure front door premium") or  (ResourceType=="CDNWEBAPPLICATIONFIREWALLPOLICIES" and "{WAFType:label}" contains "cdn")) and ("{WAF:label}" == "All" or Resource in ({WAF}))
```
(which alone is 613 ch in every query before any parameters get replaced)

to something like

```
let supportedTypes = dynamic(["APPLICATIONGATEWAYS",  "FRONTDOORS", "PROFILES", "CDNWEBAPPLICATIONFIREWALLPOLICIES"]);
let wafType = dynamic([{WAFType}]);
let wafResources = dynamic([{WAF}]);

///... other part of query...
| where ResourceType in (supportedTypes) // has_any might be even faster?
and ("*" in (wafType) or ResourceType in (wafType))
and ("*" in (wafResources ) or Resource in (wafResources))
///... other part of query

which is 344ch (without the comments, and before any parameters get replaced)

i believe this does the same thing much simpler and much more readable and easy to add to (the supported types could even be extracted out to a parameter and reused as well)